contracts/contracts/harvest/Harvester.sol

```diff
@@ -171,19 +192,16 @@ contract Harvester is Governable {
             token.safeApprove(oldUniswapAddress, 0);
         }
 
-        // Give Uniswap infinite approval when needed
```

Comment:
Is this ok? I case token still wants to use UniswapV2 this is not going to work? I would: 
- token.safeApprove(0) && token.safeApprove(MAX)(as it was before this PR)  when the token configuration passed to the harvester is using non 0 `_uniswapV2CompatibleAddr `
- token.safeApprove(0) && token.safeApprove(MAX) when uniswap v3 is configured to be used.

Reply:
Echoing what Domen said, this whole token approve section should be two independent parts. One that handles uniswap v2 and one that handles uniswap v3. Each can just separately see if it needs to operate, and does not need to know about the other.

Currently this is also missing the second to unapprove uniswapV3.